translate/storage/jsonl10n.py

```diff
@@ -416,6 +416,90 @@ def _extract_units(
             yield from parent
 
 
+class GoTextJsonUnit(BaseJsonUnit):
+    ID_FORMAT = "{}"
+
+    def getvalue(self):
+        target = self.target
+        if isinstance(target, multistring):
+            strings = list(target.strings)
+            if len(self._store.plural_tags) > len(target.strings):
+                strings += [""] * (len(self._store.plural_tags) - len(target.strings))
+            target = {
+                "select": {
+                    "feature": "plural",
+                    "arg": "N",
+                    "cases": {
+                        plural: {"msg": strings[offset]}
+                        for offset, plural in enumerate(self._store.plural_tags)
+                    },
+                }
+            }
+        value = {"id": self.getid()}
+        if self.notes:
+            value["translatorComment"] = self.notes
+        value["translation"] = target
+        return value
+
+
+class GoTextJsonFile(JsonFile):
+    """gotext JSON file
+
+    See following URLs for doc:
+
+    https://pkg.go.dev/golang.org/x/text/cmd/gotext
+    https://github.com/golang/text/tree/master/cmd/gotext/examples/extract/locales/en-US
+    """
+
+    UnitClass = GoTextJsonUnit
+
+    @property
+    def plural_tags(self):
+        return plural_tags.get(self.gettargetlanguage(), plural_tags["en"])
+
+    def _extract_units(
+        self,
+        data,
+        stop=None,
+        prev=None,
+        name_node=None,
+        name_last_node=None,
+        last_node=None,
+    ):
+        if prev is None:
+            lang = data.get("language")
+            if lang is not None:
+                self.settargetlanguage(lang)
+        for value in data["messages"]:
+            translation = value.get("translation", "")
+            if isinstance(translation, dict):
+                cases = translation.get("select", {}).get("cases", {})
+                # Ordered list of plurals
+                translation = multistring(
+                    [
+                        cases.get(key, {}).get("msg")
+                        for key in cldr_plural_categories
+                        if key in cases
+                    ]
+                )
+            unit = self.UnitClass(
+                translation,
+                value.get("id", ""),
+                value.get("translatorComment", ""),
+            )
```

Comment:
You can store placeholders on Unit this way:

```suggestion
                placeholders=value.get("placeholders", []),
            )
```

Reply:
I've started experimenting with this, but I still won't have a way to know which placeholder is the plurals one wgeb cakkubg `getvalue`, I might just remove support for now as I think this is a bit beyond my knowledge of how this library works; we can always add it back later and just have support for simple files to start, I think

Reply:
Update: pushed the various bits that can be part of message that we might want to preserve including the placeholder stuff. I feel like there *really* has to be a better way to preserve bits we don't care about for the sake of translations, but which should still be preserved across import/export of the file though?